kubernetes/process_manifest.rb

```diff
@@ -0,0 +1,157 @@
+# Copyright (C) 2017 SUSE LLC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+$LOAD_PATH.unshift File.dirname(__FILE__)
+
+require "util"
+require "yaml"
+require "fileutils"
+
+if ARGV.count != 3
+  puts "usage: #{$PROGRAM_NAME} container-manifests-dir velum-source-code-dir salt-dir"
+  Process.exit 1
+end
+
+CONTAINERS_MANIFESTS_DIR = ARGV[0]
+VELUM_SOURCE_CODE_DIR = ARGV[1]
+SALT_DIR = ARGV[2]
+
+CONTAINERS_MANIFESTS_ORIG_DIR = "/usr/share/caasp-container-manifests".freeze
+SALT_ORIG_DIR = "/usr/share/salt/kubernetes".freeze
+
+def container_volume(name:, path:)
+  {
+    "name"      => volume_name(name: name),
+    "mountPath" => path
+  }
+end
+
+def patch_container_image(container)
+  if container["image"] =~ /^sles12\/velum/
+    container["image"] = "sles12/velum:development"
+  elsif container["image"] =~ /^sles12/
+    container["image"] = "docker-testing-registry.suse.de/#{container["image"]}"
```

Comment:
i'd add some `FIXME` or `TODO` here, as we don't want the internal testing registry to stay forever.